web/src/search_pill.ts

```diff
@@ -89,45 +99,86 @@ function append_user_pill(
 
 const user_pill_operators = new Set(["dm", "dm-including", "sender"]);
 
+export function is_valid_pill(term: NarrowTerm): boolean {
+    switch (term.operator) {
+        case "has":
+            return ["image", "link", "attachment", "reaction"].includes(term.operand);
+        case "is":
+            return [
+                "dm",
+                "starred",
+                "mentioned",
+                "alerted",
+                "unread",
+                "resolved",
+                "followed",
+            ].includes(term.operand);
+        case "in":
+            return ["home", "all"].includes(term.operand);
+        case "id":
+            return message_store.get(Number.parseInt(term.operand, 10)) !== undefined;
```

Comment:
the `id:` and `near:` operators should behave identically, and just validate that they got an integer parameter. Checking `message_store` is just looking at a local cache, and thus incorrect validation -- we can't know which message IDs are valid locally. (And both are intended to allow non-existent message IDs; it's important for `near` functionally to allow stuff like `near:1` to find the start of a view)